sys/evtimer/evtimer.c

```diff
@@ -115,8 +115,10 @@ static void _update_timer(evtimer_t *evtimer)
 static uint32_t _get_offset(xtimer_t *timer)
 {
     uint64_t now_us = xtimer_now_usec64();
-    uint64_t target_us = _xtimer_usec_from_ticks64(
-                        ((uint64_t)timer->long_target) << 32 | timer->target);
+    uint64_t start_us = _xtimer_usec_from_ticks64(
+                        (_xtimer_now64() & (0XFFFFFFFF << 32)) | timer->start_time);
```

Comment:
if 64bit info is really needed, I can just provide a long_start_time member in timer.

Reply:
I think concurrency problem was already here too. If thread is descheduled for a while after now_us is assigned. now_us value does not represent the current time anymore.

Reply:
While the concurrency problem was there before we should still fix it now

Reply:
could we maybe factor out the 64bit target time into the xtimer API? ztimer_xtimer_compat would otherwise have to touch this again.
Something like `uint64_t xtimer_get_target_us(xtimer_t *t)`.

Reply:
I'd also favor that, citing myself from below:
> But from looking at this, maybe a simple (but also irq-guarded) way to get the time left on a timer would be a nice thing (?)